Java/core/src/main/java/com/amazon/randomcutforest/state/store/LeafStoreState.java

```diff
@@ -13,22 +13,29 @@
  * permissions and limitations under the License.
  */
 
-package com.amazon.randomcutforest.store;
+package com.amazon.randomcutforest.state.store;
 
 import java.util.Arrays;
 
-public class LeafStoreData {
+import lombok.Data;
 
-    public final int[] pointIndex;
-    public final short[] parentIndex;
-    public final int[] mass;
-    public final short[] freeIndexes;
+import com.amazon.randomcutforest.store.LeafStore;
 
-    public LeafStoreData(LeafStore leafStore) {
+@Data
+public class LeafStoreState {
+
+    public int[] pointIndex;
+    public short[] parentIndex;
+    public int[] mass;
+    public short[] freeIndexes;
```

Comment:
minor. this name seems inconsistent with previous variables. freeIndex?

Reply:
The current naming matches in instance variable names in the corresponding model class. `pointIndex`, `parentIndex`, and `mass` are all public fields in LeafStore, and freeIndexes is a private field in SmallIndexManager, the superclass of LeafStore.